packages/asset-swapper/src/utils/market_operation_utils/cream_utils.ts

```diff
@@ -0,0 +1,104 @@
+import { getPoolsWithTokens, parsePoolData } from 'cream-sor';
+
+import { BalancerPool } from './balancer_utils';
+
+// tslint:disable:boolean-naming
+
+interface CacheValue {
+    timestamp: number;
+    pools: BalancerPool[];
+}
+
+// tslint:disable:custom-no-magic-numbers
+const FIVE_SECONDS_MS = 5 * 1000;
+const ONE_DAY_MS = 24 * 60 * 60 * 1000;
+const DEFAULT_TIMEOUT_MS = 1000;
+const MAX_POOLS_FETCHED = 3;
+// tslint:enable:custom-no-magic-numbers
+
+export class CreamPoolsCache {
```

Comment:
Is there really any meaningful difference between this class and `BalancerPoolsCache`? Does it make sense to just modify and inherit from that class, passing in `getPoolsWithTokens()` and `parsePoolData()` callbacks to the super?